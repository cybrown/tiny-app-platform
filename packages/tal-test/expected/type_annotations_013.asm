main()
  entry:
    FunctionRef       name: a_0
    DeclareLocal      name: a, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: b_1
    DeclareLocal      name: b, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: c_2
    DeclareLocal      name: c, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Local             b
    Local             c
    Literal           2
    MakeArray
    DeclareLocal      name: d, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Local             a
    DeclareLocal      name: a2, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           0
    Local             d
    Index
    SetLocal          name: a2
a_0(row)
  entry:
    Literal           null
b_1(row)
  entry:
    Literal           null
c_2(row)
  entry:
    Literal           null
