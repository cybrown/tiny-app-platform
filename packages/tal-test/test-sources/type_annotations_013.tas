fun a(row: {name: string, age: number}) null
fun b(row: {name: string}) null
fun c(row: {age: number}) null

let d = [b, c]

var a2 = a
set a2 = d[0]
