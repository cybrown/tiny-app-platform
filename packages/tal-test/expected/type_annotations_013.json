[
  {
    "kind": "DeclareLocal",
    "mutable": false,
    "name": "a",
    "value": {
      "kind": "Function",
      "body": null,
      "parameters": [
        {
          "name": {
            "kind": "Identifier",
            "name": "row"
          },
          "type": {
            "kind": "record",
            "fields": {
              "name": {
                "kind": "named",
                "name": "string"
              },
              "age": {
                "kind": "named",
                "name": "number"
              }
            }
          }
        }
      ]
    }
  },
  {
    "kind": "DeclareLocal",
    "mutable": false,
    "name": "b",
    "value": {
      "kind": "Function",
      "body": null,
      "parameters": [
        {
          "name": {
            "kind": "Identifier",
            "name": "row"
          },
          "type": {
            "kind": "record",
            "fields": {
              "name": {
                "kind": "named",
                "name": "string"
              }
            }
          }
        }
      ]
    }
  },
  {
    "kind": "DeclareLocal",
    "mutable": false,
    "name": "c",
    "value": {
      "kind": "Function",
      "body": null,
      "parameters": [
        {
          "name": {
            "kind": "Identifier",
            "name": "row"
          },
          "type": {
            "kind": "record",
            "fields": {
              "age": {
                "kind": "named",
                "name": "number"
              }
            }
          }
        }
      ]
    }
  },
  {
    "kind": "DeclareLocal",
    "mutable": false,
    "name": "d",
    "value": {
      "kind": "Array",
      "value": [
        {
          "kind": "Local",
          "name": "b"
        },
        {
          "kind": "Local",
          "name": "c"
        }
      ]
    }
  },
  {
    "kind": "DeclareLocal",
    "mutable": true,
    "name": "a2",
    "value": {
      "kind": "Local",
      "name": "a"
    }
  },
  {
    "kind": "Assign",
    "address": {
      "kind": "Local",
      "name": "a2"
    },
    "value": {
      "kind": "Index",
      "index": 0,
      "value": {
        "kind": "Local",
        "name": "d"
      }
    }
  }
]